src/methods/chat-session-helpers.test.ts

```diff
@@ -167,4 +167,58 @@ describe("chat-session-helpers", () => {
       });
     });
   });
+  describe("isValidResponse", () => {
+    const testCases = [
+      { name: "default response", response: {}, isValid: false },
+      {
+        name: "empty candidates",
+        response: { candidates: [] } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "default candidates",
+        response: { candidates: [{}] } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "default content",
+        response: { candidates: [{ content: {} }] } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "empty parts",
+        response: {
+          candidates: [{ content: { parts: [] } }],
+        } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "default part",
+        response: {
+          candidates: [{ content: { parts: [{}] } }],
+        } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "part with empty text",
+        response: {
+          candidates: [{ content: { parts: [{ text: "" }] } }],
+        } as GenerateContentResponse,
+        isValid: false,
+      },
+      {
+        name: "part with text",
+        response: {
```

Comment:
Let's add a case where text is not defined, put a function call or something, not that it doesn't work now, but to prevent future bugs.

Reply:
added more tests